python/cudf/cudf/core/column/decimal.py

```diff
@@ -404,3 +403,7 @@ def _get_decimal_type(lhs_dtype, rhs_dtype, op):
     )
     precision = min(cudf.Decimal128Dtype.MAX_PRECISION, max_precision)
     return cudf.Decimal128Dtype(precision=precision, scale=scale)
+
+
+def _same_precision_and_scale(lhs: DecimalDtype, rhs: DecimalDtype) -> bool:
```

Comment:
Doesn't `DecimalDtype.__eq__` do this?

Reply:
Not when the class of `lhs` and `rhs` are different: (i.e., `cudf.Decimal32Dtype(3, 2) == cudf.Decimal64Dtype(3, 2) == False`)